Dockerfile

```
@@ -0,0 +1,37 @@
+# Installation stage
+# Offical Node.js Alpine base image
+FROM node:lts-alpine AS base
+
+# Set working directory
+WORKDIR /base
+
+# Copy package.json and package-lock.json for utilising Docker cache 
+# to save re-installing dependencies if unchanged
+COPY package*.json yarn.lock ./ 
+
+# Install dependencies only needed
+RUN yarn install --production --frozen-lockfile
+
+# Copy installed dependencies to current working directory
+COPY . .
+
+# Build stage
+FROM base AS builder
+ENV NODE_ENV=production
+WORKDIR /build
+COPY --from=base /base ./
+RUN yarn add --dev typescript @types/react @types/node && yarn build
+
+# Production stage
+FROM node:current-alpine AS runner
+ENV NODE_ENV=production
+WORKDIR /app
+
+# Copy files from build stage to production stage
+COPY --from=builder /build/package*.json /build/node_modules /build/.next /build/public ./
```

Quick question, couldn't we just do this? Since we're copying all the files from `/build`.
```suggestion
COPY --from=builder /build .
```

---

According to`Docker Document` you mentioned me earlier and my mentor’s suggestion, we can do like you suggested. But if concern about clean code and readability (so we know exactly which is used in production) we should keep it like this. I agree with both ways. So, what do you think?